auth/client/src/main/java/org/wildfly/security/auth/client/AuthenticationConfiguration.java

```diff
@@ -679,6 +702,51 @@ public AuthenticationConfiguration usePassword(String password) {
+     *
+     * @param maskedPasswordBytes the masked password bytes (must not be {@code null})
+     * @param algorithm the algorithm (can be {@code null})
+     * @param initialKeyMaterial the initial key material (can be {@code null})
+     * @param iterationCount the iteration count (must not be 0)
+     * @param salt the salt bytes (must not be {@code null})
+     * @param initializationVector the initialization vector (can be {@code null})
+     * @return the new configuration
+     */
+    public AuthenticationConfiguration usePassword(byte[] maskedPasswordBytes, String algorithm, char[] initialKeyMaterial, int iterationCount, byte[] salt, byte[] initializationVector) throws Exception {
+        if (iterationCount == 0) throw log.invalidIterationCount();
+        if (salt == null) throw log.invalidSalt();
+        if (maskedPasswordBytes == null) throw log.invalidMaskedPasswordBytes();
+        if (algorithm == null) algorithm = MaskedPassword.ALGORITHM_MASKED_MD5_DES;
+        if (initialKeyMaterial == null) initialKeyMaterial = "somearbitrarycrazystringthatdoesnotmatter".toCharArray();
+        final MaskedPasswordSpec spec = new MaskedPasswordSpec(initialKeyMaterial, iterationCount, salt, maskedPasswordBytes, initializationVector);
+
+        PasswordFactory factory = PasswordFactory.getInstance(algorithm);
+        MaskedPassword password = factory.generatePassword(spec).castAs(MaskedPassword.class);
+        return useMaskedPassword(password);
+    }
+
+
+    /**
+     * Create a new configuration which is the same as this configuration, but which uses the given masked password to authenticate.
+     *
+     * @param maskedPasswordBytes the masked password bytes (must not be {@code null})
+     * @param algorithm the algorithm (can be {@code null})
+     * @param initialKeyMaterial the initial key material (can be {@code null})
+     * @param iterationCount the iteration count (must not be {@code null}
+     * @param salt the salt bytes (must not be {@code null})
+     * @param initializationVector the initialization vector (can be {@code null})
+     * @return the new configuration
+     */
+    public AuthenticationConfiguration usePassword(String maskedPasswordBytes, String algorithm, String initialKeyMaterial, String iterationCount, String salt, String initializationVector) throws Exception {
+        byte[] finalMaskedPasswordBytes = maskedPasswordBytes == null ? null : CodePointIterator.ofString(maskedPasswordBytes).base64Decode().drain();
+        char[] finalInitialKeyMaterial = initialKeyMaterial == null ? null : initialKeyMaterial.toCharArray();
+        int finalIterationCount = iterationCount == null ? 0 : Integer.parseInt(iterationCount);
+        byte[] finalSalt = salt == null ? null :CodePointIterator.ofString(salt).asUtf8().drain();
+        byte[] finalInitializationVector = initializationVector == null ? null :CodePointIterator.ofString(initializationVector).base64Decode().drain();
```

Comment:
It looks like the validation checks in this method and the previous method are different but it looks like they should actually be the same.